src/Agent.Listener/DistributedTask.Pipelines/Yaml/TypeConverters/ConverterUtil.phases.cs

```diff
@@ -0,0 +1,515 @@
+
+                        case YamlConstants.EnableAccessToken:
+                            phase.EnableAccessToken = ReadNonEmptyString(parser);
+                            break;
+
+                        case YamlConstants.Target:
+                            phase.Target = ReadPhaseTarget(parser);
+                            break;
+
+                        case YamlConstants.Execution:
+                            phase.Execution = ReadPhaseExecution(parser);
+                            break;
+
+                        case YamlConstants.Variables:
+                            phase.Variables = ReadVariables(parser, simpleOnly: false);
+                            break;
+
+                        case YamlConstants.Steps:
+                            phase.Steps = ReadSteps(parser, simpleOnly: false);
+                            break;
+
+                        default:
+                            throw new SyntaxErrorException(scalar.Start, scalar.End, $"Unexpected process property: '{scalar.Value}'");
+                    }
+                }
+
+                result = phase;
+            }
+            else if (String.Equals(scalar.Value, YamlConstants.Template, StringComparison.Ordinal))
+            {
+                if (simpleOnly)
+                {
+                    throw new SyntaxErrorException(scalar.Start, scalar.End, $"A phases template cannot reference another phases '{YamlConstants.Template}'.");
+                }
+
+                var reference = new PhasesTemplateReference { Name = ReadNonEmptyString(parser) };
+                while (parser.Allow<MappingEnd>() == null)
+                {
+                    scalar = parser.Expect<Scalar>();
+                    SetProperty(parser, reference, scalar);
+                }
+
+                result = reference;
+            }
+            else
+            {
+                throw new SyntaxErrorException(scalar.Start, scalar.End, $"Unknown phase type: '{scalar.Value}'");
+            }
+
+            return result;
+        }
+
+        internal static PhaseTarget ReadPhaseTarget(IParser parser)
+        {
+            var result = new PhaseTarget();
+            if (parser.Accept<Scalar>())
+            {
+                ReadExactString(parser, YamlConstants.Server);
+                result.Type = YamlConstants.Server;
+            }
+            else
+            {
+                parser.Expect<MappingStart>();
+                while (parser.Allow<MappingEnd>() == null)
+                {
+                    Scalar scalar = parser.Expect<Scalar>();
+                    switch (scalar.Value ?? String.Empty)
+                    {
+                        case YamlConstants.DeploymentGroup:
+                            result.Type = YamlConstants.DeploymentGroup;
+                            result.Name = ReadNonEmptyString(parser);
+                            break;
+
+                        case YamlConstants.Demands:
```

Comment:
I made demands accept a string or list of strings. I'm on the fence whether this is overkill, and should just always expect a list of strings.